### PR TITLE
Design tweaks: Update icons, Use Birch colors, Typos/Formatting

### PR DESCRIPTION
**Pull request by jaryncolbert**
This PR is mostly design tweaks. I was recently reviewing the Harvest docs and noticed the favicon was still using the legacy GH logo. Since these were just updated in GHO and GHR, I took some time to update them here. 

That also reminded me that we were still using the teal color schemes from legacy GHR branding, so I updated the service docs to use the Birch color palette. 

Lastly, I noticed some small typos and formatting errors while having some of these files open, so I fixed those that I saw.

### CODE REVIEW

File: source/includes/audit-log/_introduction.md [hunk 1]
```diff
@@ -61,11 +61,11 @@ Pagination on audit log API allows for retrieving the next page from your query
 1. In the initial request:
     - Set the `paging` [query parameter](https://developers.greenhouse.io/audit-log.html#the-events-object) to `true` to receive a `paging` object with `pit_id` and `next_search_after` in the response
     - Use the `Size` header to specify the number of results you want to receive per page. The minimum value is `100`, and the maximum value is `500`
-1. In subsequent requests:
```
Comment (jsurrett) [inline source/includes/audit-log/_introduction.md:64]: While this change is fine, it is not technically needed as markdown will increase the numbers sequentially in the final document.

**Review by jsurrett — APPROVED** [review]

**Comment by jsurrett** [comment]
Do we do any accessibility testing with the color contrasts?
Screenshot 2024-10-09 at 3 25 21 PM

**Comment by jsurrett** [comment]
To be fair, our current one also fails.

File: source/includes/audit-log/_introduction.md (hunk 1 above)
Comment (jaryncolbert) [reply source/includes/audit-log/_introduction.md:64]: Agreed -- most of these formatting changes were IDE nits that didn't affect the rendering on the page. I resolved them or auto-fixed them to get rid of the yellow Rubymine warning.

**Comment by jaryncolbert** [comment]
> Do we do any accessibility testing with the color contrasts? 
Not that I know of -- I wondered the same thing. Where can I find this assessment or the source of this screenshot?

File: source/includes/gho/_api.md
Comment (malikflx) [inline source/includes/gho/_api.md:1]: The GHO changes LGTM!

**Review by malikflx — APPROVED** [review]
Approved!

**Comment by jsurrett** [comment]
@jaryncolbert - https://www.accessibilitychecker.org/

**Review by bradleyknutson — APPROVED** [review]
Looks good to me!

**Comment by jaryncolbert** [comment]
@jsurrett -- I fixed the low-hanging fruit of the missing HTML `lang` attribute, but I left the a11y colors alone for now (we're no worse off than we were contrast-wise). Playing around with the existing Birch colors, we would have to go pretty dark to be compliant, and I'd potentially want Design's input here. 

But thanks for flagging 🙏🏾

**Comment by jaryncolbert** [comment]
Got the rubber stamp from @hillrw3 on behalf of Inbound Sourcing ✅

**Pull request merged**